classes/fields/avatar.php

```diff
@@ -79,39 +79,161 @@ public function input( $name, $value = null, $options = null, $pod = null, $id =
+	 *                                  'gravatar_default' (the Gravatar logo). Default is the value of the
+	 *                                  'avatar_default' option, with a fallback of 'mystery'.
+	 *     @type bool   $force_default  Whether to always show the default image, never the Gravatar. Default false.
+	 *     @type string $rating         What rating to display avatars up to. Accepts 'G', 'PG', 'R', 'X', and are
+	 *                                  judged in that order. Default is the value of the 'avatar_rating' option.
+	 *     @type string $scheme         URL scheme to use. See set_url_scheme() for accepted values.
+	 *                                  Default null.
+	 *     @type array  $processed_args When the function returns, the value will be the processed/sanitized $args
+	 *                                  plus a "found_avatar" guess. Pass as a reference. Default null.
+	 *     @type string $extra_attr     HTML attributes to insert in the IMG element. Is not sanitized. Default empty.
+	 * }
+	 * @param mixed $id_or_email The Gravatar to retrieve. Accepts a user ID, Gravatar MD5 hash,
+	 *                           user email, WP_User object, WP_Post object, or WP_Comment object.
+	 * @return array {
+	 *     Along with the arguments passed in `$args`, this will contain a couple of extra arguments.
+	 *
+	 *     @type bool   $found_avatar True if we were able to find an avatar for this user,
+	 *                                false or not set if we couldn't.
+	 *     @type string $url          The URL of the avatar we found.
+	 * }
+	 */
+	public function get_avatar_data( $args, $id_or_email ) {
+		if ( ! $this->allow_avatar_overwrite() ) {
+			return $args;
+		}
 
-			if ( ! empty( $_user ) ) {
-				$user_id = (int) $_user->ID;
+		$return_args = $args;
+
+		$args = wp_parse_args(
+			$args,
+			array(
+				'size'           => 96,
+				'height'         => null,
+				'width'          => null,
+				'default'        => get_option( 'avatar_default', 'mystery' ),
+				'force_default'  => false,
+				'rating'         => get_option( 'avatar_rating' ),
+				'scheme'         => null,
+				'processed_args' => null, // If used, should be a reference.
+				'extra_attr'     => '',
+			)
+		);
+
+		$size = $args['size'];
+		if ( $args['width'] !== $args['height'] ) {
+			$size = $args['width'] . 'x' . $args['height'];
+		}
+
+		$user_id = $this->get_avatar_user_id( $id_or_email );
+
+		if ( 0 < $user_id && ! empty( PodsMeta::$user ) ) {
+			$user_avatar_url = null;
+			$avatar_cached   = pods_cache_get( $user_id . '-' . $size . '-url', 'pods_avatars' );
+
+			if ( ! empty( $avatar_cached ) ) {
+				$user_avatar_url = $avatar_cached;
+			} else {
+
+				$user_avatar_id = $this->get_avatar_id( $user_id );
+
+				if ( ! empty( $user_avatar_id ) ) {
+
+					$user_avatar_url = pods_image_url( $user_avatar_id, array( $args['width'], $args['height'] ), 0 );
```

Comment:
:no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 7b4cf79f730ebb8c2b39e1b5bcbbe311</sub>